### PR TITLE
[Enhancement] Validate replica number by BE count in CREATE/ALTER TABLE

### LINKED ISSUE #6619: 【show replica】Should be restricted when replication_num of changes is greater than the number of Be nodes
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
```
CREATE TABLE `aggregate_table_with_null` ( `k1` date, `k2` datetime, `k3` char(20), `k4` varchar(20), `k5` boolean, `v1` tinyint sum, `v2` smallint sum, `v3` int sum, `v4` bigint max, `v5` largeint max, `v6` float min, `v7` double min, `v8` decimal(27,9) sum ) ENGINE=OLAP AGGREGATE KEY(`k1`, `k2`, `k3`, `k4`, `k5`) COMMENT "OLAP" DISTRIBUTED BY HASH(`k1`, `k2`, `k3`, `k4`, `k5`) BUCKETS 3 PROPERTIES ( "replication_num" = "3", "storage_format" = "v2" );
```
2. ALTER TABLE '....'
```
ALTER TABLE aggregate_table_with_null SET ("replication_num" = "4")
```
3. SHOW  '....'
```
ADMIN SHOW REPLICA STATUS FROM aggregate_table_with_null;
```

### Expected behavior (Required)
Users should be prompted when the replication_num of changes is greater than the number of Be number
e.g.
```
ERROR 1064 (HY000): Failed to find enough host in all backends. need: 4, Current alive backend is [10006,321442,10007]
```
### Real behavior (Required)
alter table replication_num success，ADMIN SHOW REPLICA STATUS command execution failure 
```
mysql> ADMIN SHOW REPLICA STATUS FROM primary_table_without_null;
ERROR 2027 (HY000): Malformed packet
mysql> ADMIN SHOW REPLICA STATUS FROM primary_table_without_null;
ERROR 2013 (HY000): Lost connection to MySQL server during query
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 ```
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE d12a28a |
+----------------------+
```

### PR DESCRIPTION
**Pull request by waittttting**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6619

## Problem Summary(Required) ：

Validate replica number by BE count in CREATE/ALTER TABLE
Currently, we have the following two rules
1. replication_num needs larger than 0
2. replication_num need less than current alive be node count

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|7|7|100.00%|[]|

**Comment by padmejin** [comment]
I suggest you change the title to `validate replica number by BE count in CREATE/ALTER TABLE`

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_admit_test

**Review by padmejin — DISMISSED** [review]

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 15 / 16 (93.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|8|9|88.89%|[584]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|7|7|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -209,10 +206,19 @@ public static Short analyzeReplicationNum(Map<String, String> properties, boolea
...
+        checkTheRangeOfReplicationNum(replicationNum);
+        return replicationNum;
+    }
+
+    private static void checkTheRangeOfReplicationNum(short replicationNum) throws AnalysisException {
         if (replicationNum <= 0) {
             throw new AnalysisException("Replication num should larger than 0. (suggested 3)");
         }
-        return replicationNum;
+        List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().getBackendIds(true);
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java]: done

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 2]
```diff
@@ -209,10 +206,19 @@ public static Short analyzeReplicationNum(Map<String, String> properties, boolea
             key = PropertyAnalyzer.PROPERTIES_REPLICATION_NUM;
         }
         short replicationNum = Short.parseShort(properties.get(key));
+        checkTheRangeOfReplicationNum(replicationNum);
+        return replicationNum;
+    }
+
+    private static void checkTheRangeOfReplicationNum(short replicationNum) throws AnalysisException {
```
[the remark this answers is not in the document]
Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java]: done

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 9 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|1|2|50.00%|[597]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|7|7|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by waittttting** [comment]
run starrocks_clang-format

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|7|7|100.00%|[]|

**Comment by waittttting** [comment]
run starrocks_admit_test

**Review by gengjun-git — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 3]
```diff
@@ -209,10 +206,19 @@ public static Short analyzeReplicationNum(Map<String, String> properties, boolea
...
+    }
+
+    private static void checkAvailableBackendsIsEnough(short replicationNum) throws AnalysisException {
         if (replicationNum <= 0) {
             throw new AnalysisException("Replication num should larger than 0. (suggested 3)");
         }
-        return replicationNum;
+        List<Long> backendIds = GlobalStateMgr.getCurrentSystemInfo().getAvailableBackendIds();
+        if (replicationNum > backendIds.size()) {
+            throw new AnalysisException("Replication num should less than current alive be node count" +
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:219]: Add the replication num into the error log.
"Replication num should be less than the number of available BE nodes. Replication num is " + replication_num + " available BE nodes is " + backendIds.size();

Comment (waittttting) [reply fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java]: Done

**Comment by waittttting** [comment]
run starrocks_fe_unittest

**Comment by waittttting** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|7|7|100.00%|[]|

**Review by chaoyli — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by waittttting** [comment]
https://github.com/Mergifyio backport branch-2.4

**Pull request merged**